src/schema.js

```diff
@@ -179,6 +180,35 @@ class Schema {
     });
   }
 
+  mixinNamedParamFacade(api, def) {
+    function namedParamFacade(base, paramDef, ...args) {
+      let params = args;
+      if (args && args.length === 1 && typeof args[0] === 'object') {
+        const defaults = paramDef.reduce((result, item) => {
+          result[item.Name] = item.DefaultValue;
+          return result;
+        }, {});
+
+        const isNamedParameterObject = Object.keys(args[0]).reduce((result, key) =>
+          result && Object.prototype.hasOwnProperty.call(defaults, key), true);
+
+        if (isNamedParameterObject) {
+          Object.keys(args[0]).forEach((key) => { defaults[key] = args[0][key]; });
```

Comment:
Instead of going through `args[0]` twice, perhaps you could combine this logic with `isNamedParameterObject`?

The less loops we have, the faster we can get results :)